### ISSUE TITLE
Stopped working on iOS 12.2

### ISSUE THREAD

**Issue by macieljr** [andrastoth/webcodecamjs#80]
Accessing the demo from a device with iOS 12.2 the issue can be noticed.

I successfully changed the entry:
```
return new Promise(function (y, n) {
   (window.navigator.getUserMedia || window.navigator.mozGetUserMedia || window.navigator.webkitGetUserMedia).call(navigator, c, y, n);
});
```
To (Not the best solution... that's why I didn't forked the code to propose it. Just trying to dive deeper on the issue...):
```
return new Promise(function (y, n) {
   window.navigator.mediaDevices.getUserMedia.call(navigator, c, y, n);
});
```
This avoided the error to be issued as an alert, but something is still not working. As I'm lacking access to a Mac, I'm unable to debug the code running on the iOS device. 

Any help appreciated!

**Comment by TheAlexandru** [comment]
> return new Promise(function (y, n) {
>    window.navigator.mediaDevices.getUserMedia.call(navigator, c, y, n);
> });

This only get more errors in console :( 

I am trying to fix it as well but at the moment I have nothing...

**Comment by macieljr** [comment]
> This only get more errors in console :(

@TheAlexandru, look, I mentioned a piece of code exclusively to avoid error on iOS. Actually to avoid the error completely (also on other platforms that references 'getUserMedia' differently), I modified original code to:

```
mediaDevices.getUserMedia = function (c) {
    if (!!navigator.platform && /iPad|iPhone|iPod/.test(navigator.platform)) {
        return new Promise(function (y, n) {
            window.navigator.mediaDevices.getUserMedia.call(navigator, c, y, n);
        });
    }
    else {
        return new Promise(function (y, n) {
            (window.navigator.getUserMedia || window.navigator.mozGetUserMedia || window.navigator.webkitGetUserMedia).call(navigator, c, y, n);
        });
    }
};
```

**Comment by TheAlexandru** [comment]
> avoid error on iOS

Yes I see, I tried it and it avoided the errors and alerts but camera doesn't run anyway :(

**Comment by macieljr** [comment]
> Yes I see, I tried it and it avoided the errors and alerts but camera doesn't run anyway :(

@TheAlexandru, exactly...
Trying to figure out something but I'm lacking on resources to debug the code as the project [remotedebug-ios-webkit-adapter](https://github.com/RemoteDebug/remotedebug-ios-webkit-adapter) is used to work with to debug iOS Safari from Windows, is also not working due to the changes in iOS 12.2 as it was already registered on its issue list :(

I hope @andrastoth can help us

**Comment by dustintbrown** [comment]
Any update on this?

**Comment by TheAlexandru** [comment]
I have totally switched to quaggaJS + qrcodeJS. Didn't met any problems yet. Works as expected on any device and version.

**Comment by macieljr** [comment]
> I have totally switched to quaggaJS + qrcodeJS. Didn't met any problems yet. Works as expected on any device and version.

Unfortunately, same here...

**Comment by vidsava** [comment]
Hope to see update here.I need it work with Ipad and iphone
